packages/linter/src/index.ts

```diff
@@ -0,0 +1,136 @@
+import { cli } from "./cli";
+import { SchemaMetadataIsNews } from "./rules/SchemaMetadataIsNews";
+import { LinkRelCanonicalIsOk } from "./rules/LinkRelCanonicalIsOk";
+import { AmpVideoIsSmall } from "./rules/AmpVideoIsSmall";
+import { BookendExists } from "./rules/BookendExists";
+import { AmpVideoIsSpecifiedByAttribute } from "./rules/AmpVideoIsSpecifiedByAttribute";
+import { StoryRuntimeIsV1 } from "./rules/StoryRuntimeIsV1";
+import { StoryMetadataIsV1 } from "./rules/StoryMetadataIsV1";
+import { MetaCharsetIsFirst } from "./rules/MetaCharsetIsFirst";
+import { RuntimeIsPreloaded } from "./rules/RuntimeIsPreloaded";
+import { StoryIsMostlyText } from "./rules/StoryIsMostlyText";
+import { StoryMetadataThumbnailsAreOk } from "./rules/StoryMetadataThumbnailsAreOk";
+import { AmpImgHeightWidthIsOk } from "./rules/AmpImgHeightWidthIsOk";
+import { AmpImgAmpPixelPreferred } from "./rules/AmpImgAmpPixelPreferred";
+import { EndpointsAreAccessibleFromOrigin } from "./rules/EndpointsAreAccessibleFromOrigin";
+import { EndpointsAreAccessibleFromCache } from "./rules/EndpointsAreAccessibleFromCache";
+import { SxgVaryOnAcceptAct } from "./rules/SxgVaryOnAcceptAct";
+import { SxgContentNegotiationIsOk } from "./rules/SxgContentNegotiationIsOk";
+import { SxgDumpSignedExchangeVerify } from "./rules/SxgDumpSignedExchangeVerify";
+import { SxgAmppkgIsForwarded } from "./rules/SxgAmppkgIsForwarded";
+import { RuleConstructor } from "./rule";
+
+export enum LintMode {
+  Amp = "amp",
+  AmpStory = "ampstory",
+  Amp4Ads = "amp4ads",
+  Amp4Email = "amp4email",
+  Sxg = "sxg"
+}
+
+export enum Status {
+  PASS = "PASS",
+  FAIL = "FAIL",
+  WARN = "WARN",
+  INFO = "INFO",
+  INTERNAL_ERROR = "INTERNAL_ERROR"
+}
+
+export interface Result {
+  readonly status: Status;
+  readonly message?: string;
+}
+
+export interface Context {
+  readonly url: string;
+  readonly $: CheerioStatic;
+  readonly raw: { headers: { [key: string]: string }; body: string };
+  readonly headers: {
+    [key: string]: string;
+  };
+  readonly mode: LintMode;
+}
+
+export interface Metadata {
+  "title": string;
+  "publisher": string;
+  "publisher-logo-src": string;
+  "poster-portrait-src": string;
+  "poster-square-src"?: string;
+  "poster-landscape-src"?: string;
+}
+
+export function guessMode($: CheerioStatic): LintMode {
+  if ($("body amp-story[standalone]").length === 1) {
+    return LintMode.AmpStory;
+  }
+  // TODO Add tests for the other types
+  return LintMode.Amp;
+}
+
+function testsForMode(type: LintMode) {
+  const tests: Map<LintMode, Array<RuleConstructor>> = new Map();
+  tests.set(LintMode.Sxg, [
```

Comment:
Suggestions use the folder structure to organize and automatically import rules.